### PR TITLE
Add doc page for example of LocationComponent

### LINKED ISSUE #812: Example: using LocationComponent to show the user's current location
Create a pull requests that includes a tutorial of how to show the user's current location. This issue has an associated bounty for the [documentation](https://github.com/maplibre/maplibre/issues/191) bounty direction.

## Deliverables

- The example should cover showing the user's current location with the [`LocationComponent`](https://maplibre.org/maplibre-gl-native/android/api/-mapbox-g-l-android-s-d-k/com.mapbox.mapboxsdk.location/-location-component/index.html).
- It should be added as an activity to MapboxGLAndroidSDKTestApp (you can use an existing one if it makes sense).
- Docs should be added to MdBook in `docs/mdbook`. Use an 'anchor comment' and regular expressions to include the Kotlin source code in the docs. Explained here: https://rust-lang.github.io/mdBook/format/mdbook.html#including-portions-of-a-file
- Explain that you can configure the LocationComponent with [`LocationComponentActivationOptions.Builder`](https://maplibre.org/maplibre-gl-native/android/api/-mapbox-g-l-android-s-d-k/com.mapbox.mapboxsdk.location/-location-component-activation-options/-builder/index.html).
- The example should cover how to [change the color of the puck](https://github.com/maplibre/maplibre-gl-native/blob/01729b9a49af9836686658a525ddae3cdc6698cb/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/location/CustomizedLocationPulsingCircleActivity.kt#L286).
- Code samples should use Kotlin
- Include at least two screenshots in your tutorial (you may want to provide a fake location to not expose your location). Drag and drop them to GitHub so we don't need to store them in the git tree.
- Please don't re-use any guides or tutorials that do not have a permissive license. [This](https://github.com/aws-samples/amazon-location-samples) might be helpful, which does have a permissive license (so can be used).

### PR DESCRIPTION
**Pull request by typebrook**
This resolves #812 

- Add mdbook anchors in related activity of test app [BasicLocationPulsingCircleActivity.kt]
- Update `SUMMARY.md` for this chapter
- [x] This example covers showing the user's current location with the LocationComponent.
- [x] Explain how to configure the LocationComponent with LocationComponentActivationOptions.Builder.
- [x] Explain how to change the color of the puck.
- [x] Screenshot is attached

[BasicLocationPulsingCircleActivity.kt]: https://github.com/maplibre/maplibre-native/blob/main/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/location/BasicLocationPulsingCircleActivity.kt

### CODE REVIEW

**Comment by typebrook** [comment]
Reviewers may check results on my VPS temporarily
https://topo.tw/public/tmp/book/android/location-component-guide.html

<details>
<summary>Preview image for doc page</summary>
<image src="https://github.com/maplibre/maplibre-native/assets/19887090/8223b2c6-3fbe-40e8-ac3a-afca42648ba6"/>
</details>

File: docs/mdbook/src/android/location-component-guide.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  <!-- Always include this permission -->
+  <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
+
+  <!-- Include only if your app benefits from precise location access. -->
+  <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+</manifest>
+```
+
+2. Create a new activity named `BasicLocationPulsingCircleActivity`:
+  - This Activity Implements the `OnMapReadyCallback` interface. The `onMapReady()` method is used to manage the application once the map is ready.
```
Comment (ovivoda) [inline docs/mdbook/src/android/location-component-guide.md:23]: I would replace it with `This Activity should implement`

Comment (ovivoda) [reply docs/mdbook/src/android/location-component-guide.md:23]: The `onMapReady()` method is triggered when the map is ready to be used.

File: docs/mdbook/src/android/location-component-guide.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  - Add a variable `permissionsManager` to manage permissions.
+  - Add a variable `locationComponent` to manage user location.
+  - At the end of the `onCreate()` method, call `checkPermissions(`) to ensure that the application can access the user's location.
+
+```kotlin
+{{#include ../../../../platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/location/BasicLocationPulsingCircleActivity.kt:top}}
```
Comment (ovivoda) [inline docs/mdbook/src/android/location-component-guide.md:29]: If I preview this, it does not look right. Can we make this more user friendly?

File: docs/mdbook/src/android/location-component-guide.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+```
+
+3. In the `checkPermissions()` method, the [PermissionManager] is used to request location permissions at runtime and handle the callbacks for permission granting or rejection.
+
+   Additionally, you should pass the results of `Activity.onRequestPermissionResult()` to it.
+
+   If the permissions are granted, call `mapView.getMapAsync(this)` to execute the `onMapReady()` method when the map is ready.
```
Comment (ovivoda) [inline docs/mdbook/src/android/location-component-guide.md:36]: I would go with:

If the permissions are granted, call `mapView.getMapAsync(this)` to register the activity as a listener for `onMapReady` event.

File: docs/mdbook/src/android/location-component-guide.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+   In this demonstration, we create an instance of this class.[^1]
+
+   In this method:
+   - Use the annotation `@SuppressLint("MissingPermission")` to suppress warnings related to missing location access permissions.
+   - In `setStyle(),` you can utilize other public and token-free styles like [demotiles] instead of the [predefined styles].
+   - For the builder of [LocationComponentOptions], use `pulseEnabled(true)` to enable the pulse animation, which enhances awareness of the user's location.
+   - Use method `buildLocationComponentActivationOptions()` to set [LocationComponentActivationOptions].
+   - Use method `buildLocationComponentActivationOptions()` to set [LocationComponentActivationOptions], then activate `locatinoComponent` with it.
+   - `CameraMode.TRACKING`[^2] means that when the user's location is updated, the camera will reposition accordingly.
```
Comment (ovivoda) [inline docs/mdbook/src/android/location-component-guide.md:54]: the details here are looking good

File: docs/mdbook/src/android/location-component-guide.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+4. In the `onMapReady()` method, first set the style and then handle the user's location using the [LocationComponent].
+
+   To configure the [LocationComponent], developers should use [LocationComponentOptions].
+
+   In this demonstration, we create an instance of this class.[^1]
+
+   In this method:
+   - Use the annotation `@SuppressLint("MissingPermission")` to suppress warnings related to missing location access permissions.
+   - In `setStyle(),` you can utilize other public and token-free styles like [demotiles] instead of the [predefined styles].
+   - For the builder of [LocationComponentOptions], use `pulseEnabled(true)` to enable the pulse animation, which enhances awareness of the user's location.
```
Comment (ovivoda) [inline docs/mdbook/src/android/location-component-guide.md:51]: here I'd just go with:

Make sure you `activateLocationComponent` of `locationComponent` You can also set `locationComponent`'s various properties like `isLocationComponentEnabled` , `cameraMode` , etc...

File: docs/mdbook/src/android/location-component-guide.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+       .foregroundTintColor(Color.BLACK)  // Set color of user location
+       .build()
+```
+
+7. Here is the final results with different color configurations. For the complete content of this demo, please refer to the source code of the [Test APP] [^3].
+
+   ![result](https://github.com/maplibre/maplibre-native/assets/19887090/03dfc87b-111b-4dd0-b4a3-d89e30ed6b63)
+
+
+[^1]: [LocationComponentOptions] can also be set by defining a style in your app's `style.xml` file.
```
Comment (ovivoda) [inline docs/mdbook/src/android/location-component-guide.md:86]: I would remove this, as it is not fully detailed. It might cause more confusion than help.

**Review by ovivoda — CHANGES_REQUESTED** [review]
I left a few comments.

File: docs/mdbook/src/android/location-component-guide.md (hunk 2 above)
Comment (ovivoda) [reply docs/mdbook/src/android/location-component-guide.md:29]: hmm it looks like it looks good in the preview you provided.

File: docs/mdbook/src/android/location-component-guide.md (hunk 5 above)
Comment (typebrook) [reply docs/mdbook/src/android/location-component-guide.md:51]: How about this? c3b5f6d

```diff
    - Use method `buildLocationComponentActivationOptions()` to set [LocationComponentActivationOptions], then activate `locatinoComp
+   - To apply options, make sure you call `activateLocationComponent()` of `locationComponent`. You can also set `locationComponent`  various properties like `isLocationComponentEnabled` , `cameraMode` , etc...
    - `CameraMode.TRACKING`[^1] means that when the user's location is updated, the camera will reposition accordingly.
```

File: docs/mdbook/src/android/location-component-guide.md (hunk 6 above)
Comment (typebrook) [reply docs/mdbook/src/android/location-component-guide.md]: Yes, this makes sense.
cfda02a

File: docs/mdbook/src/android/location-component-guide.md (hunk 3 above)
Comment (typebrook) [reply docs/mdbook/src/android/location-component-guide.md]: Done in cfda02a

File: docs/mdbook/src/android/location-component-guide.md (hunk 1 above)
Comment (typebrook) [reply docs/mdbook/src/android/location-component-guide.md]: Done in cfda02a

**Comment by typebrook** [comment]
@ovivoda Preview on my VPS is updated, please visit:
https://topo.tw/p/tmp/book/android/location-component-guide.html

**Review by ovivoda — APPROVED** [review]
Thanks! It looks good!

**Comment by louwers** [comment]
@ovivoda Could you merge & comment in the issue to award the bounty?

**Comment by typebrook** [comment]
Do we use github action to deploy mdbook pages?
https://github.com/maplibre/maplibre-native/blob/main/.github/workflows/gh-pages-mdbook.yml

**Pull request merged**